front/app/containers/Admin/pagesAndMenu/containers/PagesMenu/index.tsx

```diff
@@ -1,36 +1,69 @@
-import React from 'react';
+import React, { useState } from 'react';
 
-import { WrappedComponentProps } from 'react-intl';
+import { Box, Tooltip } from '@citizenlab/cl2-component-library';
 import { Outlet as RouterOutlet } from 'react-router-dom';
 
+import useNavbarItems from 'api/navbar/useNavbarItems';
+
 import SectionFormWrapper from 'containers/Admin/pagesAndMenu/components/SectionFormWrapper';
 
 import Button from 'components/UI/Button';
 
-import { injectIntl } from 'utils/cl-intl';
+import { useIntl } from 'utils/cl-intl';
+
+import AddProjectNavbarItemModal from '../AddProjectNavbarItemModal';
 
 import messages from './messages';
 
-const PagesMenu = ({ intl: { formatMessage } }: WrappedComponentProps) => {
+const PagesMenu = () => {
+  const { data: navbarItems } = useNavbarItems();
+  const { formatMessage } = useIntl();
+  const [addProjectModalIsOpen, setAddProjectModalIsOpen] = useState(false);
+
+  if (!navbarItems) {
+    return null;
+  }
+
+  const disabledAddProjectToNavbarButton = navbarItems.data.length >= 7;
```

Comment:
Nit: I wonder if it could be valuable to have a helper method (for example a predicate) to determine if the number of items were exceeded, so it can be reused in other places, like knowing if we can add a static page.

Reply:
Good point! I don't think it's worth it's own utility but I added a constant for the max number of nav items to make it clear where this number comes from.